### PR TITLE
feat: Created a personal-info card on sidebar

### PR DESCRIPTION
**Pull request by m-code12**
Fixes: #41 

Screenshot (406)

After clicking on the card

Screenshot (407)

### CODE REVIEW

**Comment by garg3133** [comment]
Can you put some shadows around the card?

Also, I think you missed the details in the issue. Instead of displaying roll num and email, you need to put the designation of the volunteer. Please read the issue once again.

**Comment by garg3133** [comment]
Also, the user must be redirected to the profile page (`/volunteers/profile/<user_id>/`) and not the update profile page. This too was mentioned in the issue.

**Comment by m-code12** [comment]
> Also, the user must be redirected to the profile page (`/volunteers/profile/<user_id>/`) and not the update profile page. This too was mentioned in the issue.
> 
> One more thing, can you change the background of the card to white?

Yeah I'll do the changes.

**Comment by m-code12** [comment]
@garg3133 Is it correct now?

File: templates/sidebar.html [hunk 1]
```diff
@@ -110,9 +110,10 @@
             </div>
             
         </a>
-        <nav class="collapse col-md-4 col-lg-3 d-md-block bg-light sidebar" id="dash-sidebar">
+        <nav class="collapse col-md-4 col-lg-3 d-md-block bg-white sidebar" id="dash-sidebar">
```
Comment (garg3133) [inline templates/sidebar.html:113]: I just wanted the card's background to be white and not the complete background of sidebar.

File: templates/sidebar.html [hunk 2]
```diff
@@ -110,9 +110,10 @@
...
             <div class="sidebar-sticky pb-4 pb-md-0">
                 <ul class="sidebar-list nav flex-column text-center">
+                    <div class="card px-2 py-2 mt-4 shadow">
```
Comment (garg3133) [inline templates/sidebar.html:116]: This shadow looks too much, doesn't it? I think `shadow-sm` will be better here. Also, take care of the indentation here.

File: templates/sidebar.html [hunk 3]
```diff
@@ -130,15 +131,21 @@
                         <li class="nav-item">
                             <h5>Name: <span class="lead"> {{ profile.get_full_name }}</span> </h5>
                         </li>
+                    {% endif %}
```
Comment (garg3133) [inline templates/sidebar.html:134]: You don't need to end one if statement before starting the other. Nested if statements are allowed in django templates.

File: templates/sidebar.html [hunk 4]
```diff
@@ -130,15 +131,21 @@
...
+                    
+                    {% if volun.desig is None %}
                         <li class="nav-item">
-                            <h5>Roll No: <span class="lead">{{ volun.roll_no }}</span> </h5>
+                            <h5>Designation: <span class="lead">Volunteer</span> </h5>
```
Comment (garg3133) [inline templates/sidebar.html:138]: Remove the titles (like Name, Designation) and keep the values only. Also, you can remove the span element and put the values in h5 element only (removing the margins of h5). Also, having the text color of designation as red would look good, I think.

**Review by garg3133 — COMMENTED** [review]
The design of the card looks a bit off now. See if you can make it look a bit better. Also, include the email in the card below the name (and you can give some space between email and designation, it should look good that way).

File: templates/sidebar.html (hunk 3 above)
Comment (m-code12) [reply templates/sidebar.html:134]: > You don't need to end one if statement before starting the other. Nested if statements are allowed in django templates.

I tried removing it but I am getting error.

**Comment by m-code12** [comment]
> The design of the card looks a bit off now. See if you can make it look a bit better. Also, include the email in the card below the name (and you can give some space between email and designation, it should look good that way).

Just a space between email and designation isn't looking good , hence I am keeping it on separate line.

**Comment by garg3133** [comment]
No, I meant some vertical space (with margin) between email and designation.

**Comment by m-code12** [comment]
> No, I meant some vertical space (with margin) between email and designation.

Okay, I think it's looking good after making the changes. You can have a look.
Also, don't you think this is a medium level issue?

File: templates/sidebar.html (hunk 3 above)
Comment (garg3133) [reply templates/sidebar.html:134]: I'm not saying to remove the endif statement, but to place that endif statement after the designation (that is, enclose the designation also in the first if statement).

Comment (m-code12) [reply templates/sidebar.html:134]: Ohh okay

**Comment by garg3133** [comment]
> Also, don't you think this is a medium level issue?

Then how do the easy issues look like? You are finding it a bit difficult because you've never worked in Django. But, no worries, if you actually learned something from this issue, I'll assign you 50 points for this.

Coming to the changes you just made, please use proper indentations in the code (card div and li elements are on the same level and there are some indentation issues at the bottom as well. Also, you've left the Designation in the else clause and there are a few empty lines with just spaces (empty lines should be blank).

**Comment by garg3133** [comment]
One more thing, the card is not responsive to the resizing. Can you fix that too?

**Comment by m-code12** [comment]
> > Also, don't you think this is a medium level issue?
> 
> Then how do the easy issues look like? You are finding it a bit difficult because you've never worked in Django. But, no worries, if you actually learned something from this issue, I'll assign you 50 points for this.
> 
> Coming to the changes you just made, please use proper indentations in the code (card div and li elements are on the same level and there are some indentation issues at the bottom as well. Also, you've left the Designation in the else clause and there are a few empty lines with just spaces (empty lines should be blank).

Yeah I have corrected it.

**Comment by garg3133** [comment]
Why have you increased the width of sidebar? It is looking super weird now and doesn't really solve the issue. You just need to set the appropriate style to wrap down the email.

Also, please write meaningful commit messages. For now, you can squash all your commits into a single commit. Follow the below steps:
* `git rebase -i HEAD~<no. of commits you made in total>`
* Except for first `pick`, replace all `pick` with `squash` in the file opened.
* Close the file after saving it.
* In the new file opened, remove all commit messages and write a new one and then close this file too after saving.
* Force push your changes. (`git push origin +<your-brach-name>`).

**Comment by m-code12** [comment]
I have made the changes and squashed my commits.

**Comment by garg3133** [comment]
Looks good now. Although the email issue wasn't still fixed and you forgot to mention `text-danger` class in the else clause of volunteer designation, I fixed these myself. Merged this PR into master.

Thanks @m-code12 for your contribution 🚀

**Pull request merged**